sdk/core/core-http/lib/util/xml.browser.ts

```diff
@@ -147,3 +152,25 @@ function buildNode(obj: any, elementName: string): Node[] {
     throw new Error(`Illegal value passed to buildObject: ${obj}`);
   }
 }
+
+export function convertAtomXmlToJson(body: string): any {
+  const dom = parser.parseFromString(body, "text/xml");
+  throwIfError(dom);
+  const result = domToObject(dom);
+  return result;
+}
+
+/**
+ * @param {object} content The content payload as it is to be serialized. It should include any root node(s).
+ */
+export function convertJsonToAtomXml(content: any): string {
+  content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+  const res = buildNode(content, "content");
+  const dom = res[0];
+
+  return (
+    `<?xml version="1.0" encoding="utf-8" standalone="yes"?><entry xmlns="http://www.w3.org/2005/Atom"><updated>${new Date().toISOString()}</updated>` +
```

Comment:
Correct me if I am wrong here... A valid atom xml doc can either have an `entry` tag or a `feed` tag.

Here in `convertJsonToAtomXml()` we are making the assumption that the given content needs to be part of the `entry` tag. I am worrying about a scenario where in the future one would see a function named `convertJsonToAtomXml` and pass in a json object with feed representation.

In that case, we would end up wrapping the feed inside the entry tag which might result in unexpected behavior.

For our purposes, can we not have the `entry` be part of the incoming content i.e set during serialization?

Reply:
This can be taken up as part of the general improvement to design we talked about.
But since its a smaller change, updated it.